app/models/task.rb

```diff
@@ -0,0 +1,12 @@
+class Task < ApplicationRecord
+  belongs_to :project
```

Comment:
4 пункт - как раз на уровне модели нужно добавить валидацию на существования проекта у задачи, а не в контроллере.
```suggestion
  belongs_to :project, required: true
```

Reply:
Понятно, я думал belongs_to достаточно.

Reply:
если `belongs_to` есть это уже required